core/src/main/java/org/apache/carbondata/core/datamap/dev/DataMapWriter.java

```diff
@@ -32,7 +32,12 @@
   /**
    * End of block notification
    */
-  void onBlockEnd(String blockId);
+  void onBlockEnd(String blockId, String directoryPath);
+
+  /**
+   * End of block notification when index got created.
+   */
+  void onBlockEndWithIndex(String blockId, String directoryPath);
```

Comment:
Why is this method required, why not `onBlockEnd` is enough?

Reply:
onBlockEnd Method is called once the block is written. onBlockEndWithIndex is called once the index is also written after the carbondata is written out.

Reply:
I did not get the meaning of index. it is supposed to be independent of other indexes. I think onBlockEnd event is enough for writing the index file.

Reply:
But during onBlockEnd as the carbonIndex is not yet written, we wont be able to access the carbonIndex files. In the example i am gathering informations from CarbonIndex Files too. 
Better to keep hook after writing Index Files also. In future we may need some more hooks at different points.